core/rfb.js

```diff
@@ -919,6 +996,23 @@
 
             if (this._sock.rQwait("sub auth capabilities", 16 * subAuthCount, 4)) { return false; }
 
+            // Newer X10 Supermicro motherboards get here.
+            // N.B.(kelleyk): If we had trouble with this heuristic matching
+            // non-ATEN servers, we could also add the "only security type 0x10
+            // is supported" condition from above.
+            if (this._rfb_version === 3.8 &&
+                    numTunnels === 0 &&
+                    (subAuthCount === 0 || (subAuthCount & 0xFFFF) == 0x0100)) {
+                Util.Info('Detected ATEN iKVM server (using heuristic #1 -- newer AST2400 BMC?).');
+                return this._negotiate_aten_auth();
+            }
+
+            // empty sub-auth list received means 'no auth' subtype selected
+            if (subAuthCount === 0) {
```

Comment:
I don't know for a fact that it isn't necessary, and I'm nervous about fiddling with the detection heuristics and hand-shake process; getting that working properly was one of the more tedious parts of this project.

I suspect the rationale is something like "the ATEN iKVM servers do something weird and nonstandard", but let me fire up a debugger and investigate.